samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -403,12 +405,14 @@ private GrouperMetadataImpl getGrouperMetadata(String jobModelVersion, List<Proc
     Map<TaskName, List<SystemStreamPartition>> taskToSSPs = new HashMap<>();
     if (jobModelVersion != null) {
       JobModel jobModel = readJobModelFromMetadataStore(jobModelVersion);
-      for (ContainerModel containerModel : jobModel.getContainers().values()) {
-        for (TaskModel taskModel : containerModel.getTasks().values()) {
-          taskToProcessorId.put(taskModel.getTaskName(), containerModel.getId());
-          for (SystemStreamPartition partition : taskModel.getSystemStreamPartitions()) {
-            taskToSSPs.computeIfAbsent(taskModel.getTaskName(), k -> new ArrayList<>());
-            taskToSSPs.get(taskModel.getTaskName()).add(partition);
+      if (jobModel != null) {
+        for (ContainerModel containerModel : jobModel.getContainers().values()) {
+          for (TaskModel taskModel : containerModel.getTasks().values()) {
+            taskToProcessorId.put(taskModel.getTaskName(), containerModel.getId());
+            for (SystemStreamPartition partition : taskModel.getSystemStreamPartitions()) {
+              taskToSSPs.computeIfAbsent(taskModel.getTaskName(), k -> new ArrayList<>());
```

Comment:
Could this be 
`taskToSSPs.putIfAbsent(taskModel.getTaskName(), new ArrayList<>());` ?